pysumma/ensemble.py

```diff
@@ -0,0 +1,164 @@
+from copy import deepcopy
+from distributed import Client
+import pandas as pd
+import time
+import xarray as xr
+
+from .simulation import Simulation
+from .utils import ChainDict, product_dict
+
+
```

Comment:
for i, (n, s) in enumerate(e.simulations.items()):
    if s._status != 'Success':
        print(i, f'{n:100}', s._status)

When I ran this code, there was an error. 
        
0 ++fixedThermalCond_snow=0.35++                                                                       Error
1 ++fixedThermalCond_snow=0.39999999999999997++                                                        Error
2 ++fixedThermalCond_snow=0.44999999999999996++                                                        Error
3 ++fixedThermalCond_snow=0.49999999999999994++                                                        Error
4 ++fixedThermalCond_snow=0.5499999999999999++                                                         Error
5 ++fixedThermalCond_snow=0.5999999999999999++                                                         Error
6 ++fixedThermalCond_snow=0.6499999999999999++                                                         Error
7 ++fixedThermalCond_snow=0.7++                                                                        Error
8 ++fixedThermalCond_snow=0.7499999999999999++                                                         Error
9 ++fixedThermalCond_snow=0.7999999999999998++                                                         Error

Reply:
I reviewed filemanagers using `e.simulations.items()`, but it looked there was no error.

Reply:
I'm not sure what is happening without more information.

Reply:
![image](https://user-images.githubusercontent.com/23341664/55602191-64a49600-5732-11e9-9474-d6bea1b1ffaf.png)
![image](https://user-images.githubusercontent.com/23341664/55602205-7423df00-5732-11e9-9919-93124ed32776.png)
![image](https://user-images.githubusercontent.com/23341664/55602230-9289da80-5732-11e9-9c1c-02e2ec8e0260.png)

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/anaconda3/envs/pysumma/lib/python3.6/site-packages/distributed/client.py in _gather(self, futures, errors, direct, local_worker)
   1486                         try:
-> 1487                             st = self.futures[key]
   1488                             exception = st.exception

KeyError: '_submit-490b5bd250fdfdaf9c3c2f67410c25a9'

During handling of the above exception, another exception occurred:

CancelledError                            Traceback (most recent call last)
<ipython-input-28-02bf69958f4e> in <module>
      1 e.start('local', arg_list=['export OMP_NUM_THREADS=1', 
      2                            'export LD_LIBRARY_PATH=/opt/local/lib'])
----> 3 e.monitor()

~/anaconda3/envs/pysumma/lib/python3.6/site-packages/pysumma/ensemble.py in monitor(self)
    126 
    127     def monitor(self):
--> 128         simulations = self._client.gather(self.submissions)
    129         for s in simulations:
    130             self.simulations[s.run_suffix] = s

~/anaconda3/envs/pysumma/lib/python3.6/site-packages/distributed/client.py in gather(self, futures, errors, maxsize, direct, asynchronous)
   1654             return self.sync(self._gather, futures, errors=errors,
   1655                              direct=direct, local_worker=local_worker,
-> 1656                              asynchronous=asynchronous)
   1657 
   1658     @gen.coroutine

~/anaconda3/envs/pysumma/lib/python3.6/site-packages/distributed/client.py in sync(self, func, *args, **kwargs)
    674             return future
    675         else:
--> 676             return sync(self.loop, func, *args, **kwargs)
    677 
    678     def __repr__(self):

~/anaconda3/envs/pysumma/lib/python3.6/site-packages/distributed/utils.py in sync(loop, func, *args, **kwargs)
    280             e.wait(10)
    281     if error[0]:
--> 282         six.reraise(*error[0])
    283     else:
    284         return result[0]

~/.local/lib/python3.6/site-packages/six.py in reraise(tp, value, tb)
    691             if value.__traceback__ is not tb:
    692                 raise value.with_traceback(tb)
--> 693             raise value
    694         finally:
    695             value = None

~/anaconda3/envs/pysumma/lib/python3.6/site-packages/distributed/utils.py in f()
    265             if timeout is not None:
    266                 future = gen.with_timeout(timedelta(seconds=timeout), future)
--> 267             result[0] = yield future
    268         except Exception as exc:
    269             error[0] = sys.exc_info()

~/.local/lib/python3.6/site-packages/tornado/gen.py in run(self)
    727 
    728                     try:
--> 729                         value = future.result()
    730                     except Exception:
    731                         exc_info = sys.exc_info()

~/.local/lib/python3.6/site-packages/tornado/gen.py in run(self)
    740                             exc_info = None
    741                     else:
--> 742                         yielded = self.gen.send(value)
    743 
    744                 except (StopIteration, Return) as e:

~/anaconda3/envs/pysumma/lib/python3.6/site-packages/distributed/client.py in _gather(self, futures, errors, direct, local_worker)
   1491                             six.reraise(CancelledError,
   1492                                         CancelledError(key),
-> 1493                                         None)
   1494                         else:
   1495                             six.reraise(type(exception),

~/.local/lib/python3.6/site-packages/six.py in reraise(tp, value, tb)
    691             if value.__traceback__ is not tb:
    692                 raise value.with_traceback(tb)
--> 693             raise value
    694         finally:
    695             value = None

CancelledError: _submit-490b5bd250fdfdaf9c3c2f67410c25a9